src/Telemetry/AzureDevOpsTimeline/BuildLogScraper.cs

```diff
@@ -0,0 +1,50 @@
+using Microsoft.DotNet.Internal.AzureDevOps;
+using Microsoft.Extensions.Logging;
+using System;
+using System.IO;
+using System.Net.Http;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.AzureDevOpsTimeline
+{
+    public class BuildLogScraper : IBuildLogScraper
+    {
+        private readonly ILogger<BuildLogScraper> _logger;
+        private readonly IAzureDevOpsClient _azureDevOpsClient;
+
+        public BuildLogScraper(ILogger<BuildLogScraper> logger, IAzureDevOpsClient client)
+        {
+            _azureDevOpsClient = client;
+            _logger = logger;
+        }
+
+        public Task<string> ExtractMicrosoftHostedPoolImageNameAsync(string logUri, CancellationToken cancellationToken)
+            => ExtractImageNameAsync(logUri, azurePipelinesRegex, cancellationToken);
+
+        public Task<string> ExtractOneESHostedPoolImageNameAsync(string logUri, CancellationToken cancellationToken)
+            => ExtractImageNameAsync(logUri, oneESRegex, cancellationToken);
+
+        private async Task<string> ExtractImageNameAsync(string logUri, Regex imageNameRegex, CancellationToken cancellationToken)
+        {
+            var imageName = await _azureDevOpsClient.TryGetImageName(logUri, imageNameRegex, LogException, cancellationToken);
+
+            if (imageName == string.Empty)
```

Comment:
Here we would then expect `null` from the API instead of wondering whether it always returns the image name or throws (it's then easier to reason about how it behaves)